### PR TITLE
[Main Course] - Docker Windows

### PR DESCRIPTION
**Pull request by Tina0330**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+最后，重启docker desktop，你会看到成功启动的界面。
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/success.jpg" width="500" />
+  <figcaption> 成功界面 </figcaption>
+</figure>
+
+## 总结
+有些软件在电脑中的成功运行会因为系统的不兼容，需要一些先决条件。不管是参考具体的官方文档还是搜索一些避雷的经验贴，安装成功并使用后，你会发现这些环境的搭建是一劳永逸的，快尝试在你的Windows系统中运行docker吧！之后我们还会持续分享docker的学习经验，敬请期待。
```
Comment (jhqian0909) [inline docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md:87]: 感觉第一句话有点语病。Docker要大写。

**Review by jhqian0909 — CHANGES_REQUESTED** [review]
Thanks.

File: docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+template: overrides/blogs.html
+---
+
+# 教你成功在Win10系统中运行docker
+
+!!! info 
+    作者：Tina，发布于2021-11-13，阅读时间：约5分钟，微信公众号文章链接：[:fontawesome-solid-link:]()
+    
+## 1 前言
+在上文[Docker初体验](https://mp.weixin.qq.com/s/gfO5BiK9fqRtWf8rjP8mPA)中我们介绍了Docker的一些基本概念和常用命令，但因为Docker是在Linux系统下创建的资源分离机制，所以它无法在Windows系统下直接运行。这次我们将用3分钟的时间介绍一下如何在Win10系统下运行Docker。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md:11]: 引用的时候记得手动插入链接，不然会变成纯文本

File: docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+用户可以根据自己的系统在Docker官网选择[Mac](https://docs.docker.com/desktop/mac/install/ 'Install docker in Mac')或[Windows](https://docs.docker.com/desktop/windows/install/#install-docker-desktop-on-windows 'Install docker in Windows')，因为Docker可以在Mac系统中安装后可以直接运行，这里就不再赘述。
+
+安装完Docker，并注册个人账号，再双击启动它，你会发现并不像你想的那么顺利。错误信息如下图所示：
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/failed.jpg" width="500" />
+  <figcaption> Docker启动失败 </figcaption>
+</figure>
+
+这是因为Docker无法在Windows系统中直接运行，而需要创建一个Wins系统中的Linux虚拟机（VM）为Docker的正常运行搭建好底层依赖。虽然[官方文档](https://docs.docker.com/desktop/windows/install/#wsl-2-backend 'WSL2的安装')提供了hyper-V和WSL2两种解决方案，但是WSL2是基于hpyer-v在WSL1的升级版，其功能，安装方式也简单方便，因此作者就选择了WSL2的后端方式来运行Docker。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md:23]: Windows系统？不是wins系统吧？

File: docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+```shell
+wsl --set-default-version 2
+```
+
+
+## 4 检查docker设置
+成功安装好Linux distribution后，还需要在docker desktop中设置一下关于WSL2的基本参数才能确保docker的成功运行。
+
+首先，需要在通用设置下，勾选使用WSL2基础的引擎，如图所示：
```
Comment (Realvincentyuan) [inline docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md:66]: 空一行比较安全

**Review by Realvincentyuan — APPROVED** [review]
macOS YYDS

**Comment by Tina0330** [comment]
@Realvincentyuan @jhqian0909 修改了，伙伴们。谢谢！

File: docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+用户可以根据自己的系统在Docker官网选择[Mac](https://docs.docker.com/desktop/mac/install/ 'Install docker in Mac')或[Windows](https://docs.docker.com/desktop/windows/install/#install-docker-desktop-on-windows 'Install docker in Windows')，因为Docker可以在Mac系统中安装后可以直接运行，这里就不再赘述。
+
+安装完Docker，并注册个人账号，再双击启动它，你会发现并不像你想的那么顺利。错误信息如下图所示：
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/failed.jpg" width="500" />
+  <figcaption> Docker启动失败 </figcaption>
+</figure>
+
+这是因为Docker无法在Windows系统中直接运行，而需要创建一个Windows系统中的Linux虚拟机（VM）为Docker的正常运行搭建好底层依赖。虽然[官方文档](https://docs.docker.com/desktop/windows/install/#wsl-2-backend 'WSL2的安装')提供了hyper-V和WSL2两种解决方案，但是WSL2是基于hpyer-v在WSL1的升级版，其功能，安装方式也简单方便，因此作者就选择了WSL2的后端方式来运行Docker。
```
Comment (jz263) [inline docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md:25]: typo: 第二个hyper-V

**Review by jz263 — APPROVED** [review]

File: docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md (hunk 5 above)
Comment (Tina0330) [reply docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md]: 改了，谢谢嘉杰！

File: docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+这是因为Docker无法在Windows系统中直接运行，而需要创建一个Windows系统中的Linux虚拟机（VM）为Docker的正常运行搭建好底层依赖。虽然[官方文档](https://docs.docker.com/desktop/windows/install/#wsl-2-backend 'WSL2的安装')提供了hyper-V和WSL2两种解决方案，但是WSL2是基于hpyer-V在WSL1的升级版，其功能，安装方式也简单方便，因此作者就选择了WSL2的后端方式来运行Docker。
+
+## 3 安装WSL2
+
+WSL，Windows Subsystem for Linux，含义就是在Windows系统下Linux的子系统。只需要三步就可以在系统中为Docker安装WSL2了，请注意目前WSL2支持在Windows10 2004以上的版本。
+
+### 3.1 开启WSL2功能
+
+首先，打开`Powershell`命令提示符，运行命令先查看网上所有的子系统， 再选择你想要选择的系统，这里我们将选择Ubuntu来进行安装。
```
Comment (EchoTang04) [inline docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md:33]: 是否要以管理员身份运行才可以？如果是 可以强调一下。

**Review by EchoTang04 — APPROVED** [review]

File: docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md (hunk 6 above)
Comment (Tina0330) [reply docs/Main_Course/Analytics_Tools/2021-11-13-Docker-Windows.md:33]: 我直接打开就运行了，你安装的时候有提示吗，糖糖

**Review by jhqian0909 — APPROVED** [review]

**Pull request merged**